os/mm/mm_heap/mm_manage_allocfail.c

```diff
@@ -121,10 +176,24 @@ void mm_manage_alloc_fail(struct mm_heap_s *heap, int startidx, int endidx, size
 #endif /* CONFIG_MM_ASSERT_ON_FAIL */
 
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
-	for (int idx = startidx; idx <= endidx; idx++) {
-		heapinfo_parse_heap(&heap[idx], HEAPINFO_DETAIL_ALL, HEAPINFO_PID_ALL);
-	}
+	/* heapinfo_parse_heap() operations can print logs for more than 3 minutes,
+	 * which can be rebooted by watchdog. So try to stop the watchdog. */
+
+	int ret = mm_heapinfo_stop_watchdog();
+
+#ifndef CONFIG_MM_ASSERT_ON_FAIL
+	/* If CONFIG_MM_ASSERT_ON_FAIL is enabled, will be rebooted anyway, So even if
+	 * fail to stop watchdog, print heapinfo_parse_heap log. */
+
+	if (!ret)
```

Comment:
If we use OK and ERROR, then this could be changed like below
```
int ret = mm_heapinfo_stop_watchdog();
if (ret == OK) {
   print heap info
}
```
We can understand like this
1. try to stop
2. if stopped, print heapinfo

I think `!` mean negative. I could understand like below
1. try to stop
2. if it's wrong, print heapinfo

Reply:
Thank you for detail review, changed